app/src/main/java/org/wikipedia/util/UriUtil.java

```diff
@@ -120,6 +120,19 @@ public static String getUrlWithProvenance(Context context, PageTitle title,
         return title.getUri() + "?wprov=" + context.getString(provId);
     }
 
+    @NonNull
+    public static String getFilenameFromUploadUrl(@NonNull String url) {
+        String[] splitArray = url.split("/");
+        String thumbnailName = splitArray[splitArray.length - 1];
+        if (splitArray.length > 2) {
+            String originalFilename = splitArray[splitArray.length - 2];
+            if (originalFilename.matches("^[\\w,\\s-]+\\.[A-Za-z]{3}$")) {
```

Comment:
It's not at all guaranteed that the filename extension will always be three (3) characters. Simple counterexample:
https://en.wikipedia.org/api/rest_v1/page/summary/NGC_4565

I'm pretty sure the path structure of the thumbnail URL is stable, so we can probably assume that the second-to-last path segment will be the filename. (and not require any regex)

Reply:
You are right. 
Since I found that if I remove the regex, we will not get the correct filename from the second-to-last segment from an original file URL.
I have added a simpler check to check whether the URL contains `/thumb/` and extract the filename from it or not.